src/EXTRA-PAIR/pair_lj_sphere.cpp

```diff
@@ -0,0 +1,354 @@
+/* ----------------------------------------------------------------------
+   LAMMPS - Large-scale Atomic/Molecular Massively Parallel Simulator
+   https://www.lammps.org/, Sandia National Laboratories
+   LAMMPS development team: developers@lammps.org
+
+   Copyright (2003) Sandia Corporation.  Under the terms of Contract
+   DE-AC04-94AL85000 with Sandia Corporation, the U.S. Government retains
+   certain rights in this software.  This software is distributed under
+   the GNU General Public License.
+
+   See the README file in the top-level LAMMPS directory.
+------------------------------------------------------------------------- */
+
+#include "pair_lj_sphere.h"
+
+#include "atom.h"
+#include "comm.h"
+#include "error.h"
+#include "force.h"
+#include "math_special.h"
+#include "memory.h"
+#include "neigh_list.h"
+#include "neighbor.h"
+#include "update.h"
+
+#include <cmath>
+#include <cstring>
+
+using namespace LAMMPS_NS;
+using MathSpecial::powint;
+
+/* ---------------------------------------------------------------------- */
+
+PairLJSphere::PairLJSphere(LAMMPS *lmp) : Pair(lmp)
+{
+  writedata = 1;
+}
+
+/* ---------------------------------------------------------------------- */
+
+PairLJSphere::~PairLJSphere()
+{
+  if (allocated) {
+    memory->destroy(setflag);
+    memory->destroy(cutsq);
+
+    memory->destroy(cut);
+    memory->destroy(epsilon);
+  }
+}
+
+/* ---------------------------------------------------------------------- */
+
+void PairLJSphere::compute(int eflag, int vflag)
+{
+  int i, j, ii, jj, inum, jnum, itype, jtype;
+  double xtmp, ytmp, ztmp, rtmp, delx, dely, delz, evdwl, sigma6, fpair;
+  double rsq, r2inv, r6inv, forcelj, factor_lj;
+  int *ilist, *jlist, *numneigh, **firstneigh;
+
+  evdwl = 0.0;
+  ev_init(eflag, vflag);
+
+  double **x = atom->x;
+  double **f = atom->f;
+  double *radius = atom->radius;
+  int *type = atom->type;
+  int nlocal = atom->nlocal;
+  double *special_lj = force->special_lj;
+  int newton_pair = force->newton_pair;
+
+  inum = list->inum;
+  ilist = list->ilist;
+  numneigh = list->numneigh;
+  firstneigh = list->firstneigh;
+
+  // loop over neighbors of my atoms
+
+  for (ii = 0; ii < inum; ii++) {
+    i = ilist[ii];
+    xtmp = x[i][0];
+    ytmp = x[i][1];
+    ztmp = x[i][2];
+    rtmp = radius[i];
+    itype = type[i];
+    jlist = firstneigh[i];
+    jnum = numneigh[i];
+
+    for (jj = 0; jj < jnum; jj++) {
+      j = jlist[jj];
+      factor_lj = special_lj[sbmask(j)];
+      j &= NEIGHMASK;
+
+      delx = xtmp - x[j][0];
+      dely = ytmp - x[j][1];
+      delz = ztmp - x[j][2];
+      rsq = delx * delx + dely * dely + delz * delz;
+      jtype = type[j];
+
+      if (rsq < cutsq[itype][jtype]) {
+        r2inv = 1.0 / rsq;
+        r6inv = r2inv * r2inv * r2inv;
+
+        sigma6 = powint(mix_distance(rtmp, radius[j]), 6);
```

Comment:
I think if these are not diameters, then 2 particles of radius 0.5, will want to be 0.5 apart (eng minimum) instead of 1.0 apart.

Reply:
Said another way, I think if you create a system of diam=1 spheroidal particles and use this new pair style, you want it to be identical to pair lj/cut with point particles and sigma = 1 ?

Reply:
> you want it to be identical to pair lj/cut with point particles and sigma = 1 ?

My test had diameter 2.0 for that case, but I see now that I want diameter 1.0 -> sigma 1.0. So I have to apply a factor of 2.0 to the result of mix_distance() and update the documentation.

Reply:
@sjplimp Please see the revised test inputs and their outputs attached here. This is what it should be.

[in.melt-lepton.txt](https://github.com/lammps/lammps/files/11114633/in.melt-lepton.txt)
[in.melt-sphere.txt](https://github.com/lammps/lammps/files/11114635/in.melt-sphere.txt)
[in.melt.txt](https://github.com/lammps/lammps/files/11114636/in.melt.txt)
[log.melt-lepton.txt](https://github.com/lammps/lammps/files/11114637/log.melt-lepton.txt)
[log.melt-sphere.txt](https://github.com/lammps/lammps/files/11114638/log.melt-sphere.txt)
[log.melt.txt](https://github.com/lammps/lammps/files/11114639/log.melt.txt)